File: Weird_Algorithm.py
def weird_algorithm(n):
    while n > 1:
        print(n, end=' ')  # Print the current value of n without a newline
        if n % 2 == 0:
            n //= 2  # Use integer division for even n
        else:
            n = 3 * n + 1  # Calculate the new value for odd n
    print(n)  # Print the last value, which will be 1

# Read input from the user
n = int(input())
weird_algorithm(n)

File: Weird_Algorithm.java
import java.util.Scanner;

public class Weird_Algorithm {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        // Continue looping until n becomes 1
        while (n > 1) {
            System.out.printf("%d ", n); // Print the current value of n

            // Apply the weird algorithm rules
            if (n % 2 == 0) {
                n = n / 2;
            } else {
                n = (n * 3) + 1;
            }
        }

        // Print the last value, which will be 1
        System.out.printf("%d", n);

        scanner.close();
    }
}
